templates/sls-app-template-nodejs/index.js

```diff
@@ -0,0 +1,53 @@
+/**
```

Comment:
Templates  need some refactoring. serverless apps can contain multiple functions while this template is written initially to support one function per service. We might need to move the core function stuff (components, config etc.) into ./functions/$myfunction

Ex:  https://github.com/serverless/examples/tree/master/aws-golang-auth-examples (check how this service has got 4 functions defined inside a folder - `functions`). We should adjust our templates to fit into this structure (ofcourse with just one function inside it as part of the initial template).

Reply:
Well, I would be quite capable to author something for 'nodejs' and 'java' but 'go' and 'python' would sure be out of my domain. Can we create a small story regarding this task?

Reply:
Rajani's created the story: https://rally1.rallydev.com/#/184770457944d/detail/userstory/294918004464 It's appreciated a lot

Reply:
As far as I can see right now the functions under sls-app-template-node.js/components do not persist any global internal state deeming them invalid for multi-functional environment. Now it seems that the modules can be used simultaneously for as many functions as needed without any repercussions thus for 'nodejs' projects we do not need to copy the components folder inside every new function to be created. We can easily re-use the functionality inside sls-app-template-node.js/components without side effects

Reply:
That sounds correct (technically) for utils (config-handler, error-handler etc.) & it also avoids code duplication. But, we should still move index.js, tests etc to each function specific folder. 

So, something like this (?)
```
my-service/
  serverless.yml (in future..)
  deployment-env.yml
  components/
   error-handler.js
   config-handler.js
  function1/
    index.js
    config/
      dev-config.json
      prod-config.json
    test/
  function2/
    index.js
    config/
      dev-config.json
      prod-config.json
    test/
```

Reply:
Correct. **index.js**, **tests** and **config** all must move under the appropriate function folder. I'll refactor accordingly